go-app-ussd_pmtct.js

```diff
@@ -0,0 +1,267 @@
+                        self.im.set_lang(identity.extra.language_choice);
+                        self.im.user.set_answer("consent", identity.extra.consent);
+                        self.im.user.set_answer("dob", identity.extra.dob;
+
+                        // !consent_already_given
+                        if (self.im.user.consent) {
+                            return self.states.create("state_consent", dob_already_stored);
+                        }
+
+                        // !dob_already_stored
+                        if (self.im.user.dob) {
+                            return self.states.create("state_birth_year");
+                        }
+
+                        return self.states.create("state_hiv_messages");
+
+                    } else {
+                        return self.states.create("state_end_not_registered");
+                    }
+                });*/
+        });
+
+        self.add("state_end_not_registered", function(name) {
+            return new EndState(name, {
+                text: questions[name],
+                next: "state_start"
+            });
+        });
+
+        self.add("state_consent", function(name) {
+            return new ChoiceState(name, {
+                question: questions[name],
+                // error: ,
+                choices: [
+                    new Choice("yes", $("Yes")),
+                    new Choice("no", $("No"))
+                ],
+                next: function(choice) {
+                    if (choice.value === "yes") {
+                        // set consent
+                        self.im.user.set_answer("consent", "true");
+                        /*if (self.im.user.dob) {
+                            return "state_hiv_messages";
+                        } else {
+                            return "state_birth_year";
+                        }*/
+                        return "state_birth_year";
+                    } else {
+                        return "state_end_consent_refused";
+                    }
+                }
+            });
+        });
+
+        self.add("state_end_consent_refused", function(name) {
+            return new EndState(name, {
+                text: questions[name],
+                next: "state_start"
+            });
+        });
+
+        self.add("state_birth_year", function(name) {
+            return new FreeText(name, {
+                question: questions[name],
+                /*check: function(content) {
+                    return (utils.check_valid_number(content)
+                        & utils.check_number_in_range(content, "1900", utils.get_today().year)
+                },*/
+                next: function(content) {
+                    self.im.user.set_answer("dob_year", content);
+                    return "state_birth_month";
+                }
+            });
+        });
+
+        self.add("state_birth_month", function(name) {
```

Comment:
I notice that the text here doesn't align with the flow diagram - did you get content elsewhere? eg `Mar` instead of `March`.